gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/DuplicateObservationStatementMapperTest.java

```diff
@@ -149,23 +156,32 @@ public void doesntMergeObservationWhereTheDescriptionDoesntSubStringOfAnother()
                 generateLinksetComponent("ID-1")
         ));
 
-        mapper.mergeDuplicateObservationStatements(ehrExtract);
+        mapper.mergeOrRemoveDuplicateObservationStatements(ehrExtract);
 
         assertThat(firstEhrComposition(ehrExtract)).hasSize(3);
     }
 
 
     @Test
     public void doesntMergeObservationWhereTheCodesArentTheSame() {
+        Logger fooLogger = (Logger) LoggerFactory.getLogger(DuplicateObservationStatementMapper.class);
+        ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+        listAppender.start();
+        fooLogger.addAppender(listAppender);
+
         var ehrExtract = createExtract(List.of(
-                createObservation("ID-1", "101", "This is an observation which ends with ellipses..."),
-                createObservation("ID-2", "102", "This is an observation which ends with ellipses but there is more."),
-                generateLinksetComponent("ID-1")
-        ));
+            createObservation("ID-1", "101", "This is an observation which ends with ellipses..."),
+            createObservation("ID-2", "102", "This is an observation which ends with ellipses but there is more."),
+            generateLinksetComponent("ID-1")
+                                              ));
 
-        mapper.mergeDuplicateObservationStatements(ehrExtract);
+        mapper.mergeOrRemoveDuplicateObservationStatements(ehrExtract);
 
         assertThat(firstEhrComposition(ehrExtract)).hasSize(3);
+        List<ILoggingEvent> logsList = listAppender.list;
+        assertEquals(Level.INFO, logsList.get(0).getLevel());
+        assertEquals("ObservationStatement: 'ID-1' appears to have been truncated but no match was found.",
+                     logsList.get(0).getFormattedMessage());
```

Comment:
This is an [interesting thread](https://softwareengineering.stackexchange.com/questions/384164/should-log-statements-be-tested) on Stack Overflow that I found interesting regarding logging in unit tests.

I, personally, lean more towards [this response](https://softwareengineering.stackexchange.com/a/426030)!

It's mixed opinions of course, I just wanted to share this with you 😄!